src/gvars.c

```diff
@@ -107,46 +109,6 @@ static Obj * PtrGVars;
 
 static Obj TLVars;
 
-/****************************************************************************
-**
-*V  GVarLock  . . . . . . . . . . . . . . . . . .  lock for global variables
-**
-**  This lock is only needed for accessing global variables by name rather
-**  than index and to initialize copy/fopy information.
-*/
-
-static pthread_rwlock_t GVarLock;
-static void *GVarLockOwner;
-static UInt GVarLockDepth;
-
-static void LockGVars(int write) {
-  if (PreThreadCreation)
-    return;
-  if (GVarLockOwner == GetTLS()) {
-    GVarLockDepth++;
-    return;
```

Comment:
Note that this old locking code allowed for recursive locking when called from the same thread.

I thought this was irrelevant, as I saw no way this could ever be needed.

Alas, it is, as `UpdateCopyFopyInfo` also takes that lock, and it then calls `DeclareAllGVars` while holding the lock, and *that* calls `GVarName`.... *sigh*. I think at this point it's best to just excise the complete `GVarDescriptor` code; almost nothing really uses it anyway. I'll look into that.

Reply:
*sigh* it's not quite as simple. And I don't want to accidentally break HPC-GAP even more (though perhaps I should just give up on that pipe dream *double sigh*).

So, I guess I'll have to replicate the recursive locking ability